### PR TITLE
[BugFix] Fix an issue of catch exception

### PR DESCRIPTION
**Pull request by zaorangyang**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java [hunk 1]
```diff
@@ -214,7 +214,7 @@ private PProxyResult sendProxyRequest(PProxyRequest request) throws UserExceptio
                     PProxyResult result;
                     try {
                         result = future.get(Config.routine_load_kafka_timeout_second, TimeUnit.SECONDS);
-                    } catch (TimeoutException e) {
+                    } catch (Exception e) {
                         LOG.warn("failed to send proxy request to " + address + " err " + e.getMessage());
                         // When getting kafka info timed out, we tried again three times.
                         if (++retryTimes > 3 || (retryTimes + 1) * Config.routine_load_kafka_timeout_second >
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java:220]: Try when encountering any Exception?

**Review by trueeyu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java (hunk 1 above)
Comment (zaorangyang) [reply fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java:219]: > Retry when encountering any Exception?

yes

**Review by trueeyu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|1|00.00%|[216]|

File: fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java (hunk 1 above)
Comment (chaoyli) [reply fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java:219]: If is an exception about privilege, it seems no necessary to retry.

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java [hunk 2]
```diff
@@ -214,14 +214,16 @@ private PProxyResult sendProxyRequest(PProxyRequest request) throws UserExceptio
...
-                        if (++retryTimes > 3 || (retryTimes + 1) * Config.routine_load_kafka_timeout_second >
-                                                                        Config.routine_load_task_timeout_second) {
-                            throw e;
+                        if (e instanceof TimeoutException) {
+                            // When getting kafka info timed out, we tried again three times.
+                            if (++retryTimes > 3 || (retryTimes + 1) * Config.routine_load_kafka_timeout_second >
+                                                                            Config.routine_load_task_timeout_second) {
+                                throw e;
+                            }
+                            continue;
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java:225]: not continue when catch other exception?

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|5|00.00%|[217, 219, 221, 223, 225]|

**Comment by zaorangyang** [comment]
@Mergifyio rebase

**Comment by andyziye** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|5|00.00%|[217, 219, 221, 223, 225]|

**Review by chaoyli — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java (hunk 2 above)
Comment (jaogoy) [reply fe/fe-core/src/main/java/com/starrocks/common/util/KafkaUtil.java:225]: Why not directly put thoes lines in `catch(TimeoutException)` block?

**Comment by jaogoy** [comment]
@zaorangyang will this PR fix #23477?

**Pull request merged**